src/V2/LentMycWithMigration.sol

```diff
@@ -0,0 +1,925 @@
+        }
+
+        pendingRedeems = 0;
+        pendingDeposits = 0;
+
+        // Transfer assets requested by gov
+        asset.safeTransfer(msg.sender, amountToWithdraw);
+        amountDeployed += amountToWithdraw;
+
+        cycle += 1;
+
+        // Ensure after new cycle starts, enough is in contract to pay out the pending redemptions.
+        require(
+            asset.balanceOf(address(this)) >= redemptionAssets,
+            "MYC given is less than required"
+        );
+    }
+
+    /**
+     * @notice For `gov` to return MYC for lenders.
+     * @param amount Amount of MYC to return to this contract.
+     * @dev Use instead of normal ERC20 transfer so as to allow the contract.
+     */
+    function returnMyc(uint256 amount) external onlyGov {
+        asset.safeTransferFrom(msg.sender, address(this), amount);
+        amountDeployed -= amount;
+    }
+
+    /**
+     * @notice Emergency ETH withdrawal function.
+     */
+    function withdrawEth(uint256 amount) external onlyGov {
+        require(amount <= address(this).balance);
+        Address.sendValue(payable(msg.sender), amount);
+    }
+
+    /**
+     * @notice Emergency ERC20 token withdrawal function.
+     */
+    function withdrawToken(address token, uint256 amount) external onlyGov {
+        ERC20(token).safeTransfer(msg.sender, amount);
+    }
+
+    /**
+     * @dev Wrapper around `ERC20.transferFrom` which updates both `from` and `to` before receiving the transfer.
+     * @dev Basically before doing any transfers, we want to make sure that the user is in an updated state.
+     */
+    function transferFrom(
+        address from,
+        address to,
+        uint256 amount
+    ) public override returns (bool) {
+        require(!inPausedTransferMode, "Paused transfer mode");
+        updateUser(from);
+        updateUser(to);
+        bool ret = super.transferFrom(from, to, amount);
+        return ret;
+    }
+
+    /**
+     * @dev Wrapper around `ERC20.transferFrom` which updates both `from` and `to` before receiving the transfer.
+     */
+    function transfer(address to, uint256 amount)
+        public
+        override
+        returns (bool)
+    {
+        require(!inPausedTransferMode, "Paused transfer mode");
+        updateUser(msg.sender);
+        updateUser(to);
+        return _transfer(to, amount);
+    }
+
+    function _transfer(address to, uint256 amount) private returns (bool) {
+        bool ret = super.transfer(to, amount);
+        return ret;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+                        FETCHING UPDATED STATE
+    //////////////////////////////////////////////////////////////*/
+
+    /**
+     * @return The lentMYC balance of `user` after an account update.
+     */
+    function trueBalanceOf(address user) public view returns (uint256) {
+        (uint256 shareTransfer, ) = _updateUser(user);
+        return balanceOf(user) + shareTransfer;
+    }
```

Comment:
My thinking here was we don't want to update the trueBalanceOf of users when migrating, because we want to still be able to give them rewards in this contract (i.e. to compensate for the ~5-7 days into the current fortnightly cycle we will be).
Combine that with the fact that deposits/withdrawals/transfers will be paused. Does that seem reasonable to you?

Reply:
Also note that we need to start a new cycle & pause BEFORE migrating, otherwise people could have their trueBalanceOf lag behind what migrate is able to see.

Reply:
Yeah I think the pausing should cover it